File: backend/src/main/java/com/gxs/bank/repository/AccountRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.SavingsAccount;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface AccountRepository extends JpaRepository<SavingsAccount, UUID> {
    List<SavingsAccount> findByUserId(UUID userId);
    Optional<SavingsAccount> findByAccountNumber(String accountNumber);
    boolean existsByAccountNumber(String accountNumber);
    List<SavingsAccount> findByUserIdAndAccountType(UUID userId, SavingsAccount.AccountType accountType);
    List<SavingsAccount> findByStatus(SavingsAccount.Status status);
}

File: backend-python/com/gxs/bank/repository/AccountRepository.py
from com.gxs.bank.model.SavingsAccount import SavingsAccount
from com.gxs.bank.repository._base import BaseRepository


class AccountRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, SavingsAccount)

    def findByUserId(self, userId: str):
        return self.db.query(SavingsAccount).filter(SavingsAccount.userId == userId).all()

    def findByAccountNumber(self, accountNumber: str):
        return (
            self.db.query(SavingsAccount)
            .filter(SavingsAccount.accountNumber == accountNumber)
            .first()
        )

    def existsByAccountNumber(self, accountNumber: str) -> bool:
        return self.findByAccountNumber(accountNumber) is not None

    def findByUserIdAndAccountType(self, userId: str, accountType):
        return (
            self.db.query(SavingsAccount)
            .filter(SavingsAccount.userId == userId, SavingsAccount.accountType == accountType)
            .all()
        )

    def findByStatus(self, status):
        return self.db.query(SavingsAccount).filter(SavingsAccount.status == status).all()
